pkg/btree/bptree.go

```diff
@@ -40,19 +41,20 @@ func (t *BPTree) root() (*BPTreeNode, MemoryPointer, error) {
 	return root, mp, nil
 }
 
-func (t *BPTree) Find(key []byte) (MemoryPointer, bool, error) {
+func (t *BPTree) Find(key ReferencedValue) (ReferencedValue, MemoryPointer, error) {
```

Comment:
Why are we returning the `ReferencedValue` in `Find()`?

Am I wrong to assume that the key `ReferencedValue` param will be returned?

Reply:
> but the first key that matches could be []byte("hellp"), so returning the matched key provides a way to identify if the specific key searched exists or not.

Sounds good -- this means iff there exists a matched key, we'll return that key and it's `MemoryPointer
.

Will fix jsonl and csv tests